src/v/cloud_storage/cache_service.cc

```diff
@@ -550,8 +554,12 @@ ss::future<> cache::invalidate(const std::filesystem::path& key) {
       "Trying to invalidate {} from archival cache.",
       key.native());
     try {
+        auto path = (_cache_dir / key).native();
+        auto stat = co_await ss::file_stat(path);
         _access_time_tracker.remove_timestamp(key.native());
-        co_await recursive_delete_empty_directory((_cache_dir / key).native());
+        co_await recursive_delete_empty_directory(path);
+        _current_cache_size -= stat.size;
```

Comment:
`path` is always a regular file -- I rename the `recursive_delete_empty_directory` function to `delete_file_and_empty_parents` in a later commit, to make the name less misleading.

In future we should add an inode count limit that includes directories, so that we're not totally ignoring their cost.